src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -365,6 +390,84 @@ public static string RenewAzureToken(string tenantId, string clientId, string ap
             throw new KubeConfigException("Refresh not supported.");
         }
 
+#if NETSTANDARD2_0
+        /// <summary>
+        /// Implementation of the proposal for out-of-tree client
+        /// authentication providers as described here --
+        /// https://github.com/kubernetes/community/blob/master/contributors/design-proposals/auth/kubectl-exec-plugins.md
+        /// Took inspiration from python exec_provider.py --
+        /// https://github.com/kubernetes-client/python-base/blob/master/config/exec_provider.py
+        /// </summary>
+        /// <param name="config">The external command execution configuration</param>
+        /// <returns>The token received from the external commmand execution</returns>
+        public static string ExecuteExternalCommand(ExternalExecution config)
+        {
+            var execInfo = new Dictionary<string, dynamic>
+            {
+                {"apiVersion", config.ApiVersion},
+                {"kind", "ExecCredentials"},
+                {"spec", new Dictionary<string, bool>
+                {
+                    {"interactive", Environment.UserInteractive}
+                }}
+            };
+
+            var process = new Process();
+
+            process.StartInfo.Environment.Add("KUBERNETES_EXEC_INFO",
+                JsonConvert.SerializeObject(execInfo));
+            
+            if (config.EnvironmentVariables != null)
+                foreach (var configEnvironmentVariableKey in config.EnvironmentVariables.Keys)
+                    process.StartInfo.Environment.Add(key: configEnvironmentVariableKey,
+                        value: config.EnvironmentVariables[configEnvironmentVariableKey]);
+
+            process.StartInfo.FileName = config.Command;
+            if (config.Arguments != null)
+                process.StartInfo.Arguments = string.Join(" ", config.Arguments);
+            process.StartInfo.RedirectStandardOutput = true;
+            process.StartInfo.RedirectStandardError = true;
+            process.StartInfo.UseShellExecute = false;
+
+            try
+            {
+                process.Start();
+            }
+            catch (Exception ex)
+            {
+                throw new KubeConfigException($"external exec failed due to: {ex.Message}");
+            }
+
+            var stdout = process.StandardOutput.ReadToEnd();
+            var stderr = process.StandardOutput.ReadToEnd();
+            if (string.IsNullOrWhiteSpace(stderr) == false)
+                throw new KubeConfigException($"external exec failed due to: {stderr}");
+
+            // Wait for a maximum of 5 seconds, if a response takes longer probably something went wrong...
+            process.WaitForExit(5);
```

Comment:
parameter would be better

Reply:
A parameter from where? From whoever calls the function?

Reply:
A parameter from where? From whoever calls the function?

Reply:
is it ok to add to ExternalExecution or some global config?

Reply:
Don't think so as this is not documented anywhere (at least I couldn't find it).
As I've took inspiration from https://github.com/kubernetes-client/python-base/blob/master/config/exec_provider.py and they do not have a timeout period I believe leaving it with 5 seconds is more than enough.
@tg123 let me know what do you think.